src/test/java/selenium/YandexMailLoginTest.java

```diff
@@ -1,52 +1,51 @@
 package selenium;
 
 import lombok.extern.slf4j.Slf4j;
-import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.support.ui.ExpectedConditions;
 import org.openqa.selenium.support.ui.WebDriverWait;
 import org.testng.Assert;
-import org.testng.annotations.AfterClass;
 import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
-import utils.WebDriverUtils;
+import parameters.TestData;
+import testbase.TestBase;
+
+import java.time.Duration;
 
 import static constants.ByConstants.*;
-import static constants.TestConstants.*;
+import static constants.TestConstants.YANDEX_MAIL_INBOX_TITLE;
+import static constants.TestConstants.YANDEX_MAIL_URL;
 import static constants.TimeOutConstants.SHORT_TIMEOUT;
 
 @Slf4j
-public class YandexMailLoginTest {
-    private WebDriver driver;
+public class YandexMailLoginTest extends TestBase {
     private WebDriverWait wait;
 
     @BeforeClass
-    void setUp() {
-        driver = WebDriverUtils.setWebDriver();
-        wait = new WebDriverWait(driver, SHORT_TIMEOUT);
+    public void setUp1() {
+        wait = new WebDriverWait(driver(), SHORT_TIMEOUT);
+        driver().manage().timeouts().implicitlyWait(SHORT_TIMEOUT); //2.Add implicit waiter for WebDrive
     }
 
-    @Test
-    void TestYandexMailLogin() {
+    @Test(dataProvider = "yandexMailCredentials", dataProviderClass = TestData.class)
+    public void TestYandexMailLogin(String username, String password) throws InterruptedException {
         log.info("Navigate to the login main page");
-        driver.get(YANDEX_MAIL_URL);
-        driver.findElement(LOGIN_BUTTON_MAIN_PAGE).click();
+        driver().get(YANDEX_MAIL_URL);
+        driver().findElement(LOGIN_BUTTON_MAIN_PAGE).click();
 
         log.info("Enter username and click login button");
-        driver.findElement(EMAIL_FIELD).sendKeys(YANDEX_MAIL_USERNAME);
-        driver.findElement(LOGIN_BUTTON_AUTHORIZATION_PAGE).click();
+        driver().findElement(EMAIL_FIELD).sendKeys(username);
+        driver().findElement(LOGIN_BUTTON_AUTHORIZATION_PAGE).click();
+        Thread.sleep(3000); //3.Add Thread.sleep for login test. What type of waiter is it - Implicit waiter?
```

Comment:
It is not implicit nor explicit, since Thread.sleep() does not wait for any specific condition to be met, nor does it interact with the DOM or elements in any way.